src/ucs/sys/topo.h

```diff
@@ -0,0 +1,100 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_TOPO_H
+#define UCS_TOPO_H
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <ucs/type/status.h>
+#include <limits.h>
+
+BEGIN_C_DECLS
+
+/** @file topo.h */
+
+/**
+ * @ingroup UCS_RESOURCE
+ * System Device abstraction
+ */
+typedef struct ucs_sys_device {
+    unsigned int id;             /**< Index of the unit */
+    unsigned int bus_id;         /**< bus ID of of the device if applicable*/
+    unsigned int numa_node;      /**< NUMA node assoicated with the device*/
+    char         name[PATH_MAX]; /**< Name of the device*/
```

Comment:
could be const char *. for API simplicity
we can allocate strings internally and point to them